Nodejs/Product/Npm/SPI/NpmGetCatalogCommand.cs

```diff
@@ -65,115 +65,182 @@ internal class NpmGetCatalogCommand : NpmCommand, IPackageCatalog {
+                    return;
+                }
+
+                // Every property in the object should be either the "_updated" value, or a package
+                if(jsonReader.TokenType != JsonToken.PropertyName) {
+                    throw new JsonException("Unexpected JSON token in NPM catalog data");
+                }
+                string propertyName = (string)jsonReader.Value;
+
+                // If it's "_updated", update the revision info.
+                if(propertyName.Equals("_updated", StringComparison.Ordinal)) {
+                    jsonReader.Read();
+                    db.InsertOrReplace(new RegistryInfo() {
+                        RegistryUrl = registryUrl,
+                        Revision = (long)jsonReader.Value,
+                        UpdatedOn = DateTime.Now
+                    });
+                    continue;
+                }
+
+                // Else the property value should be an object literal representing the package
+                jsonReader.Read();
+                if (jsonReader.TokenType == JsonToken.StartObject) {
+                    IPackage package = ReadPackage(jsonReader, builder);
+                    if (package != null) {
+                        InsertCatalogEntry(db, package);
+                    }
+                }
+                else {
+                    throw new JsonException("Unexpected JSON token reading a package from the NPM catalog data");
+                }
+            }
+            throw new JsonException("Unexpected end of stream reading the NPM catalog data object");
+        }
+
+        private void ReadPackagesFromArray(SQLiteConnection db, JsonTextReader jsonReader) {
+            // Inside the array, each object is an NPM package
+            var builder = new NodeModuleBuilder();
+            while (jsonReader.Read()) {
+                switch (jsonReader.TokenType) {
+                    case JsonToken.StartObject:
+                        IPackage package = ReadPackage(jsonReader, builder);
+                        if (package != null) {
+                            InsertCatalogEntry(db, package);
+                        }
+                        break;
+                    case JsonToken.EndArray:
+                        // Done. This is the spot the function should always exit on valid data
+                        return;
+                    default:
+                        throw new JsonException("Unexpected JSON token in NPM catalog data array");
+                }
+            }
+            throw new JsonException("Unexpected end of stream reading the NPM catalog data array");
+        }
+
+        private IPackage ReadPackage(JsonTextReader jsonReader, NodeModuleBuilder builder) {
+            builder.Reset();
+
+            // The JsonTextReader should be positioned at the start of the object literal token for the package
+            var module = JToken.ReadFrom(jsonReader);

```

Comment:
Okay, we should just make sure it still works with vs2013 or remove the conditional altogether and stick wth newtonsoft v6 everywhere (it's been battle tested enough now). There are subtle differences in handling between the two versions and I'm sort of surprised it's both reading through the endobject with ReadFrom, even though we've already read the startobject